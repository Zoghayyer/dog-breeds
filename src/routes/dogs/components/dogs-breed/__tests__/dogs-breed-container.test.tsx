import React from 'react';
import { shallow } from 'enzyme';
import {
  DispatchProps,
  DogsBreedContainer,
  OwnProps,
} from '../dogs-breed-container';

describe('(Container) DogsBreedContainer', () => {
  let component;
  const defaultProps: DispatchProps & OwnProps = {
    dogs: [],
    favoriteToggle: () => ({}),
    isLoading: false,
  };

  const shallowDogsBreedContainer = (extendedProps: Partial<DispatchProps & OwnProps> = {}) => {
    component = shallow(<DogsBreedContainer {...defaultProps} {...extendedProps} />);
  };

  beforeEach(() => shallowDogsBreedContainer());

  it('renders DogsBreedView component', () => {
    expect(component).toMatchSnapshot();
  });

  it('renders the loader component if isLoading it true', () => {
    shallowDogsBreedContainer({ isLoading: true });
    expect(component).toMatchSnapshot();
  });

  describe('(Method) handleClickFavorite', () => {
    it('calls favoriteToggle with the correct id', () => {
      const favoriteToggleSpy = jest.fn();
      const props = {
        ...defaultProps,
        favoriteToggle: favoriteToggleSpy,
      };
      component = new DogsBreedContainer(props);

      component.handleClickFavorite(1);

      expect(favoriteToggleSpy).toBeCalledWith(1);
    });
  });
});
