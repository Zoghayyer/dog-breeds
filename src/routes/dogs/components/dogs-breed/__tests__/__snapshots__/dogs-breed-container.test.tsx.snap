// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Container) DogsBreedContainer renders DogsBreedView component 1`] = `
<DogsBreedView
  dogs={Array []}
  handleClickFavorite={[Function]}
/>
`;

exports[`(Container) DogsBreedContainer renders the loader component if isLoading it true 1`] = `
<Loader
  className="mt-4"
  size="sm"
/>
`;
